### PR TITLE
Fix: #534 Added tests for Symlink using Promises API

### PR DESCRIPTION
**Pull request by Pooch11**
Added tests for Symlink using promises workflow (Should expect same results as fs.Symlink with callback)
Not sure if these should be included in their own file because they are related to promises or if all symlink tests belong in the fs.symlink.spec.js file
Passed Tests

### CODE REVIEW

**Comment by Pooch11** [comment]
Linked to Issue #534

File: tests/spec/fs.symlink.spec.js [hunk 1]
```diff
@@ -43,4 +43,69 @@ describe('fs.symlink', function() {
...
   });
+  describe('fsPromises.symlink', function () {
+    it('should return an error if destination path does not exist', function () {
+      var fsPromises = util.fs().promises;
+      return fsPromises.symlink('/', '/tmp/link')
+        .catch(error => {
+          expect(error).to.exist;
+          expect(error.code).to.equal('ENOENT');
+        });
+    });
```
Comment (jagmeetb) [inline tests/spec/fs.symlink.spec.js]: Add a line of whitespace between each test method.

File: tests/spec/fs.symlink.spec.js [hunk 2]
```diff
@@ -43,4 +43,69 @@ describe('fs.symlink', function() {
...
+    it('should return an error if source path does not exist', function () {
+      var fsPromises = util.fs().promises;
+      return fsPromises.symlink('/tmp/myLink', '/myLink')
+        .catch(error => {
+          expect(error).to.exist;
+          expect(error.code).to.equal('ENOENT');
+        });
+    });
+
+
```
Comment (jagmeetb) [inline tests/spec/fs.symlink.spec.js]: You can remove one of these lines.

File: tests/spec/fs.symlink.spec.js [hunk 3]
```diff
@@ -43,4 +43,69 @@ describe('fs.symlink', function() {
...
+            .then(stats => {
+              expect(stats).to.exist;
+              expect(stats.type).to.equal('DIRECTORY');
+            })
+            .catch(error => {
+              expect(error).not.to.exist;
+              expect(error.code).to.equal('ENOENT');
+            });
+        });
+    });
```
Comment (jagmeetb) [inline tests/spec/fs.symlink.spec.js]: Add a line after this test.

File: tests/spec/fs.symlink.spec.js [hunk 4]
```diff
@@ -43,4 +43,69 @@ describe('fs.symlink', function() {
       });
     });
   });
```
Comment (jagmeetb) [inline tests/spec/fs.symlink.spec.js]: Add a comment to denote the start of fsPromises tests. For example:

```

/**
 * fsPromises tests
 */

```

File: tests/spec/fs.symlink.spec.js [hunk 5]
```diff
@@ -43,4 +43,69 @@ describe('fs.symlink', function() {
...
+            });
+        });
+    });
+    it('Promise should create a symlink of type FILE when file provided', function () {
+      var fsPromises = util.fs().promises;
+      var fs = util.fs();
+      fs.open('/myFile', 'w+', function (error) {
+        if (error) throw error;
+        expect(error).not.to.exist;
+      });
```
Comment (jagmeetb) [inline tests/spec/fs.symlink.spec.js]: Use fsPromises for the `fs.open()` method used here.

File: tests/spec/fs.symlink.spec.js [hunk 6]
```diff
@@ -43,4 +43,69 @@ describe('fs.symlink', function() {
...
+            .then(stats => {
+              expect(stats).to.exist;
+              expect(stats.type).to.equal('FILE');
+            })
+            .catch(error => {
+              expect(error).not.to.exist;
+              expect(error.code).to.equal('ENOENT');
+            });
+        });
+    });
```
Comment (jagmeetb) [inline tests/spec/fs.symlink.spec.js]: Add a line of whitespace here.

File: tests/spec/fs.symlink.spec.js [hunk 7]
```diff
@@ -43,4 +43,69 @@ describe('fs.symlink', function() {
...
+      var fsPromises = util.fs().promises;
+
+      return fsPromises.symlink('/tmp', '/')
+        .catch(error => {
+          expect(error).to.exist;
+          expect(error.code).to.equal('EEXIST');
+        });
+    });
+  });
+
```
Comment (jagmeetb) [inline tests/spec/fs.symlink.spec.js]: Remove this whitespace.

**Comment by jagmeetb** [comment]
Good job!

**Comment by humphd** [comment]
Fixed, rebased, and merged in https://github.com/filerjs/filer/commit/33339b81d757823d69d9f4cce3db52611772dcc6.

**Pull request closed without merging**